nf_core/pipelines/containers_utils.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import re
+from pathlib import Path
+
+import yaml
+
+from nf_core.utils import NF_INSPECT_MIN_NF_VERSION, check_nextflow_version, pretty_nf_version, run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class ContainerConfigs:
+    """Generates the container configuration files for a pipeline.
+
+    Args:
+        workflow_directory (str | Path): The directory containing the workflow files.
+        org (str): Organisation path.
+    """
+
+    def __init__(
+        self,
+        workflow_directory: str | Path = ".",
+        org: str = "nf-core",
+    ):
+        self.workflow_directory = Path(workflow_directory)
+        self.org: str = org
+
+    def generate_container_configs(self) -> None:
+        """Generate the container configuration files for a pipeline."""
+        self.check_nextflow_version_sufficient()
+        default_config = self.generate_default_container_config()
+        self.generate_all_container_configs(default_config)
+
+    def check_nextflow_version_sufficient(self) -> None:
+        """Check if the Nextflow version is sufficient to run `nextflow inspect`."""
+        if not check_nextflow_version(NF_INSPECT_MIN_NF_VERSION):
+            raise UserWarning(
+                f"To use Seqera containers Nextflow version >= {pretty_nf_version(NF_INSPECT_MIN_NF_VERSION)} is required.\n"
+                f"Please update your Nextflow version with [magenta]'nextflow self-update'[/]\n"
+            )
+
+    def generate_default_container_config(self) -> str:
+        """
+        Generate the default container configuration file for a pipeline.
+        Requires Nextflow >= 25.04.4
+        """
+        log.debug("Generating container config file with [magenta bold]nextflow inspect[/].")
+        try:
+            # Run nextflow inspect
+            executable = "nextflow"
+            cmd_params = f"inspect -format config {self.workflow_directory}"
+            cmd_out = run_cmd(executable, cmd_params)
+            if cmd_out is None:
+                raise UserWarning("Failed to run `nextflow inspect`. Please check your Nextflow installation.")
```

Comment:
I have a feeling we have this multiple times in our code base. should this be a utils function?

Reply:
`run_cmd` is the utils function, we raise a warning depending on where we use this function, in some cases we might not want to stop execution. I would say it's fine as it is now

Reply:
I was more thinking about https://github.com/nf-core/tools/pull/3954/changes#diff-8bdd161b806e417fa6f155d1babd55464cbb9cea5f3449392fb23e518b7ca417R363-R369